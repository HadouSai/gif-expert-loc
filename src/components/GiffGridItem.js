import React from "react";
import PropTypes from "prop-types";

/**
 *
 * @param {*} img
 * @returns
 */
const GiffGridItem = ({ id, title, imgUrl }) => {
  return (
    <li className="card" key={id}>
      <span>{title}</span>

      <img src={imgUrl} alt={title} />
    </li>
  );
};

GiffGridItem.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  imgUrl: PropTypes.string.isRequired,
};

export default GiffGridItem;
