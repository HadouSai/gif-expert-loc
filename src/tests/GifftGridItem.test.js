import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import GiffGridItem from "../components/GiffGridItem";

describe('Pruebas en GiffGridItem', () => {
    const text = "aa"
    const imgurl = "https://www.giffgrid.com/";
    const wrapper = shallow(<GiffGridItem id="1" title={text} imgUrl={imgurl} />);

    test('should renderize GiffGridItem compnent', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('should show text', () => {
        const p = wrapper.find('span');
        expect(p.text().trim()).toBe(text);
    })

    test('should show img', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(imgurl);
    })


})



