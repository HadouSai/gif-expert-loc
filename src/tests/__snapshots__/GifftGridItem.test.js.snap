// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GiffGridItem should renderize GiffGridItem compnent 1`] = `
<li
  className="card"
  key="1"
>
  <span>
    aa
  </span>
  <img
    alt="aa"
    src="https://www.giffgrid.com/"
  />
</li>
`;
